pkg/config/expand.go

```diff
@@ -630,6 +638,17 @@ func (ref varReference) ToModuleID() string {
 	return ref.toModuleID
 }
 
+func (ref varReference) HclString() string {
+	switch ref.toGroupID {
+	case globalGroupID:
+		return "var." + ref.name
+	case ref.fromGroupID:
+		return "module." + ref.toModuleID + "." + ref.name
+	default:
+		return "var." + ref.name + "_" + ref.toModuleID
+	}
```

Comment:
I feel like we lost some context with this function. 1. being what these cases stand for (global, within group, inter group) and 2. why the hcl is the way that it is. Part of me want to see the `IsIntergroup` function used but it does not lend itself to switch and the switch is also nice.

What about just adding some comments:
```
	case globalGroupID:
		return "var." + ref.name
	case ref.fromGroupID: 
		// intragroup reference can make direct reference to module output
		return "module." + ref.toModuleID + "." + ref.name
	default:  
		// for intergroup reference ghpc will create a variable
		return "var." + ref.name + "_" + ref.toModuleID
```

Reply:
Addressed on HEAD of PR.